### PR TITLE
Created the all(array, value) function.

### LINKED ISSUE #4: Check if all elements equal a specific value
```js
import { all } from 'if.js';

let array = [0,1,2,3];
all(array, 0); // false
```

### LINKED ISSUE #5: Check if all elements verify a function
```js
import { all } from 'if.js';

let array = [0,1,2,3];
all(array, x => x === 0); // false
```

### LINKED ISSUE #6: Add support for objects in `all` checker
```js
import { all } from 'if.js';

let object = { x : 1, y : 1} ;
all(object, 1); // true
```

### PR DESCRIPTION
**Pull request by Argetan**
fixes #4
fixes #5
fixes #6

I created a method that checks if all objects in an array match a certain value using `===`. Also wrote the corresponding tests and updated the `README.md`.

### CODE REVIEW

**Comment by Argetan** [comment]
Extended functionality to include #5.

**Comment by Argetan** [comment]
Now includes the functionality described in #6.

**Comment by joeltankam** [comment]
Thanks for your pull request :heart_eyes: 
I'll check it out as soon as possible

File: src/is.js [hunk 1]
```diff
@@ -125,6 +125,33 @@ export function isWindowObject(value) {
     return value != null && typeof value === 'object' && 'setInterval' in value;
 }
 
+/**
+ * Returns whether all values in the 
+ * array/object are equal to a given 
+ * value
+ * @param {*} value the array or object to check
```
Comment (joeltankam) [inline src/is.js]: Please rename `value` to `enumerable`

File: src/is.js [hunk 2]
```diff
@@ -125,6 +125,33 @@ export function isWindowObject(value) {
...
+                if (!other(value[i])) return false;
+            }
+        } else {
+            for (var i = 0; i < value.length; i++) {
+                if (value[i] !== other) return false;
+            }
+        }
+
+        return true;
+    } else if (isObject(value)) {
```
Comment (joeltankam) [inline src/is.js]: `else`'s are not needed because of the return statement

File: src/is.js [hunk 3]
```diff
@@ -125,6 +125,33 @@ export function isWindowObject(value) {
...
 }
 
+/**
+ * Returns whether all values in the 
+ * array/object are equal to a given 
+ * value
+ * @param {*} value the array or object to check
+ * @param {*} other the value or function to match
+ */
+export function matchesAll(value, other) {
```
Comment (joeltankam) [inline src/is.js]: Please rename `other` to a more significant identifier. 
I suggest `matcher`. Feel free to suggest something better.

File: src/is.js [hunk 4]
```diff
@@ -142,4 +169,5 @@ export default {
     regexp: isRegexp,
     undefined: isUndefined,
     windowObject: isWindowObject,
+    all: matchesAll,
```
Comment (joeltankam) [inline src/is.js]: `all` is not intended to be part of `is` checker. But rather a different checker, as described in the issue.

**Comment by Argetan** [comment]
I'll get to work on it later today!

**Comment by Argetan** [comment]
Alright, should be all good by the way.

**Comment by joeltankam** [comment]
Ok, but I need to setup the release process before merging any PR. Also, I'll apply some refactoring on your code after merging, I'll let you know. 
For the next time please make sure you're using our `.editorconfig` file (for example, you don't have new line at the end of your files) and `eslint` (To get feedback on code style before pushing, ie before Codacy Checks). You can find more info in the `CONTRIBUTING.md` file.

**Pull request merged**